### PR TITLE
Convert to HTTPS-only plugin

### PR DESCRIPTION
**Pull request by wolf31o2**
This converts the upstream heroku_syslog_http input plugin into a
new standalone heroku_http plugin. This is a first attempt at cleaning
up the code and making it more special purpose.

- Remove in_heroku_syslog.rb
- Rename in_heroku_syslog_http.rb to in_heroku_http.rb
- Rename logplex.rb to parser_logplex.rb
- Remove tag from configuration parameters
- Remove king from logplex parser
- Update tests
- Run through rubocop to clean up syntax

Signed-off-by: Chris Gianelloni <cgianelloni@applause.com>

### CODE REVIEW

File: lib/fluent/plugin/parser_logplex.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+module Fluent
+  module Plugin
+    class Logplex < Parser
+      HTTPS_REGEXP = /^([0-9]+)\s+\<(?<pri>[0-9]+)\>[0-9]* (?<time>[^ ]*) (?<drain_id>[^ ]*) (?<ident>[a-zA-Z0-9_\/\.\-]*) (?<pid>[a-zA-Z0-9\.]+)? *- *(?<message>.*)$/
```
Comment (ghost) [inline lib/fluent/plugin/parser_logplex.rb:7]: Can you provide an example of what you're trying to match here?

**Review by ghost — DISMISSED** [review]
Looks good overall, though I'd like to see more comments to help explain what each class/method/etc. is supposed to do.

File: lib/fluent/plugin/parser_logplex.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            m = line.match(expression)
+
+            m.names.each_with_object({}) do |name, record|
+              record[name] = m[name]
+
+              next unless name == 'pri'
+              pri = m[name].to_i
+              record['pri'] = pri
+              record['facility'] = FACILITY_MAP[pri >> 3]
+              record['priority'] = PRIORITY_MAP[pri & 0b111]
```
Comment (agaffney) [inline lib/fluent/plugin/parser_logplex.rb]: You should create constants with meaningful names for these magic numbers

File: test/plugin/test_in_heroku_http.rb [hunk 3]
```diff
@@ -155,24 +155,21 @@ def test_ignore_unmatched_drain_id
...
-      "Logplex-Msg-Count" => messages.length.to_s,
-      "Logplex-Frame-Id" => "09C557EAFCFB6CF2740EE62F62971098",
-      "Logplex-Drain-Token" => "d.fc6b856b-3332-4546-93de-7d0ee272c3bd",
-      "User-Agent" => "Logplex/v49"
-    })
+    http = Net::HTTP.new('127.0.0.1', PORT)
+    req = Net::HTTP::Post.new('/heroku', 'Content-Type' => 'application/logplex-1',
+                                         'Logplex-Msg-Count' => messages.length.to_s,
+                                         'Logplex-Frame-Id' => '09C557EAFCFB6CF2740EE62F62971098',
+                                         'Logplex-Drain-Token' => 'd.fc6b856b-3332-4546-93de-7d0ee272c3bd',
```
Comment (agaffney) [inline test/plugin/test_in_heroku_http.rb:170]: You should create constants with meaningful names for the above 2 values. Also, reading the doc link above, it seems that `Drain-Token` should be unique for the log drain plugin, and `Frame-Id` should be unique per request (for later tracking/debug purposes)

File: lib/fluent/plugin/parser_logplex.rb (hunk 1 above)
Comment (wolf31o2) [reply lib/fluent/plugin/parser_logplex.rb:7]: Sure. This is inherited code, though.

File: test/plugin/test_in_heroku_http.rb (hunk 3 above)
Comment (wolf31o2) [reply test/plugin/test_in_heroku_http.rb:170]: I'm not going to create constants for just the tests.

Comment (agaffney) [reply test/plugin/test_in_heroku_http.rb:170]: Fair enough. I didn't notice this was in the tests.

File: lib/fluent/plugin/parser_logplex.rb (hunk 2 above)
Comment (wolf31o2) [reply lib/fluent/plugin/parser_logplex.rb]: Done.

**Comment by wolf31o2** [comment]
BTW, I know this doesn't include documentation. I plan on doing that in a separate request, once this has been approved and stabilized.

**Review by agaffney — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**